### PR TITLE
Proxy Support from CLI #84

### PR DESCRIPTION
**Pull request by HaydenMeloche**
This PR is referencing #84

## What I've done
- Added CLI param to support passing proxy URL
- Refractored variable name from `proxy` to `path` as my new addition is referring to proxy as well and I don't want others confused

## What I would like to discuss
- My param currently is `-pr` or `--proxy`. Is this good or should it be changed?
- Is my variable refactor OK?
- How do I write a test for this? See line 94 in test file.
- Is it possible to test this locally without being behind a proxy?

### CODE REVIEW

**Comment by HaydenMeloche** [comment]
@JasonEtco Mind taking a look? :)

**Comment by toplac** [comment]
@tcbyrd We would love to see this support in smee-client. Whats your opinion on this?

File: client/bin/smee.js [hunk 1]
```diff
@@ -10,6 +10,7 @@ program
   .option('-t, --target <target>', 'Full URL (including protocol and path) of the target service the events will forwarded to. Default: http://127.0.0.1:PORT/PATH')
   .option('-p, --port <n>', 'Local HTTP server port', process.env.PORT || 3000)
   .option('-P, --path <path>', 'URL path to post proxied requests to`', '/')
+  .option('-pr, --proxy <proxy>', 'URL to proxy (Typically useful behind corporate firewall)')
```
Comment (JasonEtco) [inline client/bin/smee.js]:
```suggestion
  .option('-pr, --proxy <proxy>', 'URL to proxy (typically useful behind a corporate firewall)')
```

**Comment by Dancho80** [comment]
From our Debug session, it looks like NodeJS is using an invalid request when using HTTPS, specifically NodeJS is trying to send an HTTP GET request for  https://smee.io/XYZ, which includes the full path (not just the domain), which is an invalid request when using HTTPS because it exposes the path data to interception, preventing the proxy from properly securing the connection, and forcing the proxy to reject the connection with an HTTP 400 / Invalid request. While this connection string is valid for HTTP, it is not valid for HTTPS.

**Pull request closed without merging**